cmd/client.go

```diff
@@ -22,8 +22,36 @@ type ClientOptions struct {
 
 type ClientFactory func(opts ClientOptions) *fn.Client
 
-func NewClientFactory(transport http.RoundTripper) ClientFactory {
-	return func(clientOptions ClientOptions) *fn.Client {
+// NewDefaultClientFactory returns a function that creates instances of function.Client and a cleanup routine.
+//
+// This function may allocate resources that are used by produced instances of function.Client.
+//
+// To free these resources (after instances of function.Client are no longer in use)
+// caller of this function has to invoke the cleanup routine.
+//
+// Usage:
+//   newClient, cleanUp := NewDefaultClientFactory()
+//   defer cleanUp()
+//   fnClient := newClient()
+//   // use your fnClient here...
+func NewDefaultClientFactory() (newClient ClientFactory, cleanUp func() error) {
+
+	var transportOpts []fnhttp.Option
+	var additionalCredLoaders []creds.CredentialsCallback
+
+	switch {
+	case openshift.IsOpenShift():
```

Comment:
Just a comment:
I can see a future where we have a set of supported platforms, which are each given the opportunity to self-identify and add to the features offered by the Function Client.  This would work a bit like a connection upgrade attempt.  This keeps the core library clean of any third-party dependencies, but allows for our Functions library to offer the ability to seamelssly switch between cluster flavors.